### PR TITLE
Update machine controller

### PR DESCRIPTION
**Pull request by mrIncompetent**
**What this PR does / why we need it**:
Update machine controller to v0.10.2

**Release note**:
```release-note
Update machine controller to v0.10.2
```

### CODE REVIEW

File: api/cmd/kubermatic-api/main.go
Comment (alvaroaleman) [inline api/cmd/kubermatic-api/main.go:139]: Finally!

File: api/Gopkg.toml
Comment (alvaroaleman) [inline api/Gopkg.toml]: ?

**Review by alvaroaleman — COMMENTED** [review]
Can you also please remove the code that creates a temporary clientset in the ipam controller tests? We don't need that anymore

File: api/Gopkg.toml
Comment (mrIncompetent) [reply api/Gopkg.toml]: Removed with https://github.com/kubermatic/kubermatic/pull/2473/commits/a9c342ec603c21589de2e22e78c40a8529b3f746

**Comment by mrIncompetent** [comment]
> Can you also please remove the code that creates a temporary clientset in the ipam controller tests? We don't need that anymore

Done with https://github.com/kubermatic/kubermatic/pull/2473/commits/0aca9c0ddc613566042a4ddbd607b0fcf8e1b00b

File: api/pkg/crd/kubermatic/v1/sshkeys.go
Comment (p0lyn0mial) [inline api/pkg/crd/kubermatic/v1/sshkeys.go:17]: is this correct resource name ?

Comment (mrIncompetent) [reply api/pkg/crd/kubermatic/v1/sshkeys.go:17]: Yes, you can see that in the diff of the generated client. The resource name does not change.

Comment (p0lyn0mial) [reply api/pkg/crd/kubermatic/v1/sshkeys.go:17]: oh, I misread the intention of this pull because I thought that we actually want to change the name.

File: api/hack/update-codegen.sh
Comment (mrIncompetent) [inline api/hack/update-codegen.sh:19]: @alvaroaleman 
Not pretty - but works. Should have been my first idea...

File: api/Gopkg.toml
Comment (alvaroaleman) [inline api/Gopkg.toml]: Uhm why the cluster-api override, shouldnt this be made explicit by the machine-controller import? If not, we must add a check that this is in sync with what the machine-controller imports

Comment (mrIncompetent) [reply api/Gopkg.toml]: Without dep was not able to find the correct version :/
I'll check if that is still the case

Comment (mrIncompetent) [reply api/Gopkg.toml]: Fixed with https://github.com/kubermatic/kubermatic/pull/2473/commits/9f50b954d8c925099cf436559a1652773e21a6a9

File: api/Gopkg.toml
Comment (alvaroaleman) [inline api/Gopkg.toml]: Maybe remove it and not just comment out? :)

**Comment by alvaroaleman** [comment]
`pkg/controller/ipam/controller_test.go` still contains the `go generate` and cleanup code for the temp client

**Comment by mrIncompetent** [comment]
> pkg/controller/ipam/controller_test.go still contains the go generate and cleanup code for the temp client

Removed with https://github.com/kubermatic/kubermatic/pull/2473/commits/28f98b51f92d460840f1eff02192e2d0f707f3b5

**Comment by alvaroaleman** [comment]
Pull looks fine to me there is only one thing that I find irritating: I started off with the same goal before and for my PR back then https://github.com/kubermatic/kubermatic/pull/2354/files the diff is `+3,482 −188,658`, for this its a lot bigger and removing a lot less code: `+172,768 −289,417 `

Any idea what could be the reason for this?

**Comment by mrIncompetent** [comment]
> Pull looks fine to me there is only one thing that I find irritating: I started off with the same goal before and for my PR back then https://github.com/kubermatic/kubermatic/pull/2354/files the diff is +3,482 −188,658, for this its a lot bigger and removing a lot less code: +172,768 −289,417
> 
> Any idea what could be the reason for this?

We update basically the whole k8s.io/* libraries here

**Review by alvaroaleman — APPROVED** [review]

**Comment by mrIncompetent** [comment]
@p0lyn0mial PTAL

**Comment by alvaroaleman** [comment]
In the cluster-api repo is an open PR that updates its dependencies to Kubernetes 1.12: https://github.com/kubernetes-sigs/cluster-api/pull/630/files#diff-bd247e83efc3c45ae9e8c47233249f18

This means we will soon have to move the machine-controller and then Kubermatic to 1.12 as well. Maybe wait until the pull above is merged and then directly go to 1.12 in order to not have 1.11 in the gist history?

**Comment by mrIncompetent** [comment]
@p0lyn0mial Ping

**Comment by p0lyn0mial** [comment]
only `3,274` files to review . . .

**Review by p0lyn0mial — APPROVED** [review]
going through individual commits instead of the list of files was a good idea.

**Comment by alvaroaleman** [comment]
@p0lyn0mial With the [Better Pull Request for Chrome](https://chrome.google.com/webstore/detail/better-pull-request-for-g/nfhdjopbhlggibjlimhdbogflgmbiahc) extension you can have a tree view on the left and then only look at files that are not below `vendor`  :)

**Comment by p0lyn0mial** [comment]
> @p0lyn0mial With the Better Pull Request for Chrome extension you can have a tree view on the left and then only look at files that are not below vendor :)

@alvaroaleman installing :)

**Pull request merged**
